### ISSUE TITLE
sql: remove statement field from SHOW JOBS

### ISSUE THREAD

**Issue by rytaft** [cockroachdb/cockroach#36376]
The `statement` field was added to the output of `SHOW JOBS` in order to show the original SQL statement that was used for automatic statistics jobs. However, now that automatic statistics have been removed from the output of `SHOW JOBS` (visible with `SHOW AUTOMATIC JOBS` instead), we should remove the `statement` field from `SHOW JOBS`. 

cc @RaduBerinde, @Amruta-Ranade

**Comment by RaduBerinde** [comment]
I'm not sure.. Maybe all jobs should have a user-friendly description. I think it's fine to keep around in case we do that.

By the way, is there a variation of `SHOW JOBS` that would show the auto stats jobs? Do we want one?

**Comment by rytaft** [comment]
Ok! I opened based on discussion with Amruta last week, but happy to close it if we don't want to make the change.

`SHOW AUTOMATIC JOBS` shows auto stats jobs

**Comment by RaduBerinde** [comment]
Then would `SHOW JOBS` and `SHOW AUTOMATIC JOBS` have different schema? That would be a bit of a pain to document.

**Comment by rytaft** [comment]
@Amruta-Ranade implied that `SHOW AUTOMATIC JOBS` might be documented on a different page, but I'm not sure what she decided. I don't feel strongly either way.

**Comment by Amruta-Ranade** [comment]
Yes, I created a separate page for `SHOW AUTOMATIC JOBS`. Will open a PR shortly.
